[doc][core][compiled graphs] Add visualization doc (#49722)
ray-project/ray: doc/source/ray-core/compiled-graph/visualization.rst

Visualization
=============

Ray Compiled Graph provides visualization functionalities to help users with development and debugging.

Graph structure
---------------

To visualize the graph structure, call the ``visualize`` method after calling ``experimental_compile``
on the DAG.

.. testcode::

> 

Reviewer: Can you also set the env var `RAY_CGRAPH_VISUALIZE_SCHEDULE=1` in the script?

Author: hmm I think it's better to mention it later when we talk about execution schedule, rather than having it when describing the graph structure visualization.

Reviewer: In that case, how about adding a Python script with the environment variable set or at least

```sh
RAY_CGRAPH_VISUALIZE_SCHEDULE=1 python3 compiled_graph.py
```

Author: done

---------------

To visualize the graph structure, call the ``visualize`` method after calling ``experimental_compile``
on the graph.

.. testcode::

    import ray
    from ray.dag import InputNode, MultiOutputNode

    @ray.remote
    class Worker:
        def inc(self, x):
            return x + 1

        def double(self, x):
            return x * 2

        def echo(self, x):
            return x

    sender1 = Worker.remote()
    sender2 = Worker.remote()
    receiver = Worker.remote()

    with InputNode() as inp:
        w1 = sender1.inc.bind(inp)
        w1 = receiver.echo.bind(w1)
        w2 = sender2.double.bind(inp)
        w2 = receiver.echo.bind(w2)
        dag = MultiOutputNode([w1, w2])

    compiled_dag = dag.experimental_compile()
    compiled_dag.visualize()

By default, Ray generates a PNG image named ``compiled_graph.png`` and saves it in the current working directory.
Note that this requires ``graphviz`` to be installed.

The visualization for the preceding code is shown below:

.. image:: ../../images/compiled_graph.png
    :alt: Visualization of Graph Structure
    :align: center

Note that tasks of the same actor are shown in the same color.

> Execution Schedule

Reviewer 2: ```suggestion
Execution schedule
```

------------------

> Compiled Graph generates an execution schedule for each of the participating actors. This is very useful information

Reviewer 2: ```suggestion
Compiled Graph generates an execution schedule for each of the participating actors. This schedule is useful information
```

> to understand how the graph is executed at runtime.

Reviewer 2: ```suggestion
for understanding how Ray executes the graph at runtime.
```

To visualize the execution schedule, set the environment variable ``RAY_CGRAPH_VISUALIZE_SCHEDULE=1``
when running the script. For example, save the preceding code in a script named ``example.py``,
and run the following command:

.. testcode::

    RAY_CGRAPH_VISUALIZE_SCHEDULE=1 python3 example.py

> When ``experimental_compile`` is executed, Ray generates a PNG image named ``compiled_graph_schedule.png`` and

Reviewer 2: ```suggestion
When Ray executes ``experimental_compile``, it generates a PNG image named ``compiled_graph_schedule.png`` and
```
